solc/CommandLineInterface.cpp

```diff
@@ -161,7 +161,11 @@ static bool coloredOutput(CommandLineOptions const& _options)
 
 void CommandLineInterface::handleBinary(string const& _contract)
 {
-	solAssert(m_options.input.mode == InputMode::Compiler || m_options.input.mode == InputMode::CompilerWithASTImport, "");
+	solAssert(
+		m_options.input.mode == InputMode::Compiler ||
+		m_options.input.mode == InputMode::CompilerWithASTImport ||
+		m_options.input.mode == InputMode::CompilerWithEvmAssemblyJsonImport, ""
```

Comment:
A message is now optional with `solAssert()` macro.
```suggestion
		m_options.input.mode == InputMode::CompilerWithEvmAssemblyJsonImport
```

Reply:
Will do the changes regarding the c++ code, after the import/export script seem to be ready.